modules/pubmatic/openwrap/applovinmax.go

```
@@ -0,0 +1,237 @@
+	if sdkSource == nil || len(sdkSource.Ext) == 0 {
+		return
+	}
+
+	if maxRequest.Source == nil {
+		maxRequest.Source = &openrtb2.Source{}
+	}
+
+	maxRequest.Source.Ext = setIfKeysExists(sdkSource.Ext, maxRequest.Source.Ext, "omidpn", "omidpv")
+}
+
+func updateUser(sdkUser *openrtb2.User, maxRequest *openrtb2.BidRequest) {
+	if sdkUser == nil {
+		return
+	}
+
+	if maxRequest.User == nil {
+		maxRequest.User = &openrtb2.User{}
+	}
+
+	if sdkUser.Yob != 0 {
+		maxRequest.User.Yob = sdkUser.Yob
+	}
+
+	if sdkUser.Gender != "" {
+		maxRequest.User.Gender = sdkUser.Gender
+	}
+
+	if sdkUser.Keywords != "" {
+		maxRequest.User.Keywords = sdkUser.Keywords
+	}
+
+	maxRequest.User.Data = sdkUser.Data
+	maxRequest.User.Ext = setIfKeysExists(sdkUser.Ext, maxRequest.User.Ext, "consent", "eids")
+}
+
+func setIfKeysExists(source []byte, target []byte, keys ...string) []byte {
+	oldTarget := target
+	for _, key := range keys {
+		field, dataType, _, err := jsonparser.Get(source, key)
+		if err != nil {
+			continue
+		}
+
+		if len(target) == 0 {
+			target = []byte(`{}`)
+		}
+
+		if dataType == jsonparser.String {
+			quotedStr := strconv.Quote(string(field))
+			field = []byte(quotedStr)
+		}
+
+		target, err = jsonparser.Set(target, field, key)
+		if err != nil {
+			return oldTarget
+		}
+	}
+	return target
+}
+
+func updateMaxAppLovinRequest(rctx models.RequestCtx, requestBody []byte) []byte {
+	signalData := getSignalData(requestBody)
+	if signalData == nil {
+		return requestBody
+	}
+
+	maxRequest := &openrtb2.BidRequest{}
+	if err := json.Unmarshal(requestBody, maxRequest); err != nil {
+		return requestBody
+	}
+
+	if clientConfigFlag, err := jsonparser.GetInt(signalData.Ext, "wrapper", "clientconfig"); err == nil {
+		rctx.ClientConfigFlag = int(clientConfigFlag)
+	}
+
+	addSignalDataInRequest(signalData, maxRequest, rctx.ClientConfigFlag)
+
+	maxRequestbytes, _ := json.Marshal(maxRequest)
```

dont ignore this error

---

return original requestbody in case of error